### PR TITLE
hb-1

### PR DESCRIPTION
**Pull request by julia-shvabovich**

### CODE REVIEW

File: src/main/java/cinema/dao/impl/MovieDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class MovieDaoImpl implements MovieDao {
+    @Override
+    public Movie add(Movie movie) {
+        Transaction transaction = null;
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
```
Comment (andromash) [inline src/main/java/cinema/dao/impl/MovieDaoImpl.java]: According to the checklist we need to use try-finally and close session manually

File: src/main/java/cinema/dao/impl/MovieDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+            transaction = session.beginTransaction();
+            Long id = (Long) session.save(movie);
+            transaction.commit();
+            movie.setId(id);
```
Comment (andromash) [inline src/main/java/cinema/dao/impl/MovieDaoImpl.java]: setId is redundant in this case, Hibernate does it automatically

**Review by andromash — CHANGES_REQUESTED** [review]
good job!

File: src/main/java/cinema/dao/impl/MovieDaoImpl.java (hunk 2 above)
Comment (julia-shvabovich) [reply src/main/java/cinema/dao/impl/MovieDaoImpl.java]: wow!

File: src/main/java/cinema/dao/impl/MovieDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+            }
+        }
+    }
+
+    @Override
+    public List<Movie> getAll() {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<Movie> movies = session.createQuery("from Movie", Movie.class);
+            return movies.getResultList();
+        }
```
Comment (dmytro-hryhoruk) [inline src/main/java/cinema/dao/impl/MovieDaoImpl.java:44]: let's add catch block here too

File: src/main/java/cinema/service/impl/MovieServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+import cinema.lib.Inject;
+import cinema.lib.Service;
+import cinema.model.Movie;
+import cinema.service.MovieService;
+import java.util.List;
+
+@Service
+public class MovieServiceImpl implements MovieService {
+    @Inject
+    MovieDao movieDao;
```
Comment (dmytro-hryhoruk) [inline src/main/java/cinema/service/impl/MovieServiceImpl.java]: access modifier

Comment (julia-shvabovich) [reply src/main/java/cinema/service/impl/MovieServiceImpl.java]: done

File: src/main/java/cinema/dao/impl/MovieDaoImpl.java (hunk 3 above)
Comment (julia-shvabovich) [reply src/main/java/cinema/dao/impl/MovieDaoImpl.java:44]: done

File: src/main/java/cinema/util/HibernateUtil.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    private static SessionFactory sessionFactory = initSessionFactory();
+
+    private HibernateUtil() {
+    }
+
+    private static SessionFactory initSessionFactory() {
+        try {
+            return new Configuration().configure().buildSessionFactory();
+        } catch (Exception e) {
+            throw new DataProcessingException("Couldn't create session factory", e);
```
Comment (dmytro-hryhoruk) [inline src/main/java/cinema/util/HibernateUtil.java:17]: in this case we should use different type of exception compare to Dao layer, let's use just Runtime here

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**